Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,94 @@
 import React from 'react';
 import './App.css';
+import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
+import ButtonsContainer from './components/ButtonComponents/ButtonsContainer';
 
+/*
 const App = () => {
   return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
+    <div className='calculator-container'>
+      <CalculatorDisplay value="0" />
+      
+      <ButtonsContainer />
     </div>
   );
 };
 
 export default App;
+*/
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: "",
+      operator: "",
+      memory: "",
+    };
+  }
+
+  clickNumber(button) {
+    this.setState({
+      total: (this.state.total + button.target.dataset.value)
+    })
+  }
+
+  clearFunction() {
+    this.setState({
+      total: "",
+      operator: "",
+      memory: ""
+    })
+  }
+
+  clickOperator(button) {
+    this.setState({
+      operator: button.target.dataset.value,
+      memory: this.state.total,
+      total: ""
+    });
+  }
+
+  calculateResult() {
+    switch (this.state.operator) {
+      case '+':
+        this.setState({
+          total: (Number(this.state.memory) + Number(this.state.total))
+        });
+        break;
+      case '-':
+        this.setState({
+          total: (Number(this.state.memory) - Number(this.state.total))
+        });
+        break;
+      case 'x':
+        this.setState({
+          total: (Number(this.state.memory) * Number(this.state.total))
+        });
+        break;
+      case '÷':
+        this.setState({
+          total: (Number(this.state.memory) / Number(this.state.total))
+        });
+        break;
+      default:
+        break;
+    }
+  }
+
+  render() {
+    return (
+      <div className='calculator-container'>
+        <CalculatorDisplay total={this.state.total} />
+        
+        <ButtonsContainer
+         clickFunction={this.clickNumber.bind(this)} 
```

Comment:
That could be refactored with Arrow functions to decrease the need of binding so many times. Overall Arrow Functions are the safest and most efficient way to `bind` this i believe.